app/modules/site_settings/models.py

```diff
@@ -517,37 +476,265 @@ def get_edm_configuration(cls, key, default=None, **kwargs):
+        return returned_data
+
+    @classmethod
+    def _get_houston_definition(cls, key):
+        key_conf = cls._get_houston_setting_conf(key)
+        is_public = key_conf.get('public', True)
+        data = {
+            'descriptionId': f'CONFIGURATION_{key.upper()}_DESCRIPTION',
+            'labelId': f'CONFIGURATION_{key.upper()}_LABEL',
+            'defaultValue': '',
+            'isPrivate': not is_public,
+            'settable': True,
+            'required': True,
+            'fieldType': 'string',
+            'displayType': 'string',
+        }
+
+        # Some variables have specific values so incorporate those as required
+        extra_definition = key_conf.get('definition', None)
+        if extra_definition:
+            data.update(extra_definition)
         return data
 
     @classmethod
-    def post_edm_configuration(cls, path, kwargs):
+    def get_all_rest_definitions(cls):
+        houston_definitions = {}
+        for key in cls.get_setting_keys():
+            houston_definitions[key] = cls._get_houston_definition(key)
+
+        data = {'response': {'configuration': houston_definitions}}
+        if not is_extension_enabled('edm'):
+            return data
+
+        # Add all the EDM configuration data
+        edm_definitions = cls._get_edm_definitions()
+        edm_definitions['response']['configuration'].update(
+            data['response']['configuration']
+        )
+        data = edm_definitions
+
+        # Populate site.species suggested values from the ia_config
+        species_json = data['response']['configuration']['site.species']
+
+        if species_json is not None:
+            from app.modules.ia_config_reader import IaConfig
+
+            ia_config_reader = IaConfig()
+            species = ia_config_reader.get_configured_species()
+            if not isinstance(species_json['suggestedValues'], list):
+                species_json['suggestedValues'] = ()
+            for (
+                scientific_name
+            ) in species:  # only adds a species that is not already in suggestedValues
+                needed = True
+                for suggestion in species_json['suggestedValues']:
+                    if suggestion.get('scientificName', None) == scientific_name:
+                        needed = True
+                if needed:
+                    details = ia_config_reader.get_frontend_species_summary(
+                        scientific_name
+                    )
+                    if details is None:
+                        details = {}
+                    species_json['suggestedValues'].insert(
+                        0,
+                        {
+                            'scientificName': scientific_name,
+                            'commonNames': [details.get('common_name', scientific_name)],
+                            'itisTsn': details.get('itis_id'),
+                        },
+                    )
+
+        # TODO also traverse private here FIXME
```

Comment:
@naknomum, This was a hanging comment in the resources file. Is this still relevant?